.azure/pr-check.yaml

```
@@ -23,5 +30,39 @@ steps:
       python .azure/scripts/validate_samples.py
     displayName: "Metadata Validation"
 
+  # Give execute permissions to trivy-scan.sh
+  - script: |
+      echo "Giving execute permission to trivy-scan.sh"
+      chmod +x .azure/scripts/trivy-scan.sh
+    displayName: "Set execute permissions"
+
+  - template: install/install-trivy.yaml@common-templates
+  - script: |
+        echo "##[group]Running Trivy scan for each image"
+
+        # Initialize exit_code as 0 (success)
+        exit_code=0
+
+        # Read image_urls.txt and pass each image to trivy-scan.sh
+        while IFS= read -r image; do
+          echo "Scanning image: $image"
+          .azure/scripts/trivy-scan.sh "$image"
+
+          # Capture the exit code from the Trivy scan
+          scan_exit_code=$?
+
+          # If the scan fails (non-zero exit code), set exit_code to non-zero
+          if [ $scan_exit_code -ne 0 ]; then
+            echo "Trivy scan failed for image: $image"
+            exit_code=1  # Mark exit_code as failure
+          fi
+        done < image_urls.txt
+
+        echo "##[endgroup]"
+
+        # Exit with the appropriate exit code to fail the step if any scan failed
+        exit $exit_code
+    displayName: "Run Trivy scan for all images"
```

I think it's better to say
```suggestion
    displayName: "Run Trivy scan for quick deployable images"
```
WDYT?

---

+1